parts/k8s/windowsconfigfunc.ps1

```diff
@@ -41,6 +41,28 @@ function Install-Docker
         $DockerVersion
     )
 
+    # Note: to get a list of all versions, use this snippet
+    # $versions = (curl.exe -L "https://go.microsoft.com/fwlink/?LinkID=825636&clcid=0x409" | ConvertFrom-Json).Versions | Get-Member -Type NoteProperty | Select-Object Name
+    # Docker version to API version decoder: https://docs.docker.com/develop/sdk/#api-version-matrix
+
+    switch ($DockerVersion.Substring(0,5))
+    {
+        "17.06" {
+            Write-Log "Docker 17.06 found, setting DOCKER_API_VERSION to 1.30"
+            [System.Environment]::SetEnvironmentVariable('DOCKER_API_VERSION', '1.30', [System.EnvironmentVariableTarget]::Machine)

```

Comment:
In 17.06 1.30 is the default api version.  Is this check necessary?

Reply:
This is actually a key read by the kubelet. The kubelet doesn't set the version used in the client today, so this is the way to force it.

Reply:
It is working in 17.06 via https://github.com/Azure/acs-engine/pull/4119.  Not sure I understand why the extra environment variable is needed in this case?